src/utils.ts

```diff
@@ -6,6 +6,52 @@
  */
 import { ContextAttributes } from './constructor-types';
 
+function chceksum(obj: any): string {
+  const str = JSON.stringify(obj);
+  let hash = 0;
+  for (let i = 0; i < str.length; i++) {
+    const chr = str.charCodeAt(i);
+    hash = (hash << 5) - hash + chr;
+    hash |= 0;
+  }
+  return hash.toString(16);
+}
+
+/**
+ * a class that hold an array of objects that each object would be unique
+ *
+ */
+class ObjectSet {
+  dataArray: any[] = [];
```

Comment:
Usually we use 'unknown' type in order to declare a variable that we don't know its type.
'any' is not recommended by typescript and should be used only when you have no other choice.
There difference is that typescript compiler forces you to cast unknown types to something you are familiar with, and in that sense it is more safe.